### PR TITLE
Call remaining liquidation auction entrypoints in end to end tests

### PR DESCRIPTION
**Pull request by dorranh**
Adds calls to the remaining liquidation auction entrypoints which were previously not called in our end to end tests (#189). To allow this to run in a reasonable amount of time, I had to update our e2e patch to modify the `max_bid_interval_in_blocks` and `max_bid_interval_in_seconds` to much smaller values. Otherwise the test would have to wait up to 20 minutes for the auction to complete.

### CODE REVIEW

File: e2e/main.py [hunk 1]
```diff
@@ -294,22 +326,89 @@ def call_bulk(bulks, *, batch_size):
...
+            ](),
+        ):
+            # Only retrieve the first queued slice ptr for efficiency
+            break
+        # Deposit a ton of tez to this burrow to ensure that it is no longer liquidatable
+        call_endpoint(
+            checker,
+            "deposit_tez",
+            leaf["leaf"]["value"]["contents"]["burrow"][1],
+            amount=100_000_000_000,
```
Comment (gkaracha) [inline e2e/main.py:362]: Ha, that's a ton of tez indeed!

File: e2e/main.py [hunk 2]
```diff
@@ -294,22 +326,89 @@ def call_bulk(bulks, *, batch_size):
...
+        for leaf_ptr, leaf in auction_avl_leaves(checker, current_auctions_ptr):
+            # Double check that we are only working with leaf ptrs
+            assert "leaf" in leaf
+            auctioned_slices.append(leaf_ptr)
+
+        # Note: batching here based on empirical estimates of how many slices we can
+        # touch in one go before hitting the gas limit.
+        for i in range(0, len(auctioned_slices), 10):
+            slices_to_cancel = auctioned_slices[i : i + 10]
+            call_endpoint(checker, "touch_liquidation_slices", slices_to_cancel)
```
Comment (gkaracha) [inline e2e/main.py:407]: I wonder if we'd gain anything if we touched the slices a little out of order here, to cover more execution paths. It's certainly not necessary; just a thought :slightly_smiling_face:

**Review by gkaracha — APPROVED** [review]
Nice, LGTM!

File: e2e/main.py (hunk 1 above)
Comment (dorranh) [reply e2e/main.py:362]: :moneybag:

File: e2e/main.py (hunk 2 above)
Comment (dorranh) [reply e2e/main.py:407]: That's a good point. I think that ties in well with the goal of adding some profiling to these tests, so I noted it in a new issue here: https://github.com/tezos-checker/checker/issues/194.

Comment (gkaracha) [reply e2e/main.py:407]: Nice! :+1:

**Comment by utdemir** [comment]
I'm a bit concerned on us looking at the underlying storage to get some information in order to write this test case; I feel like it would be more realistic if we were only using the public interface (entrypoints and views) to achieve this. 

It probably is the case where we don't expose enough views to achieve this. I guess:

* The winner of an auction has to have a way to see/touch the slices for their lot.
* A burrow owner has to be able to see/touch the slices for their burrow.

There might be more than one possble interface to achieve the above.

I guess, since we are just trying to measure the gas costs it might be fine; but still if some of the things we are doing are natural things to do on checker, we should have a nice way of doing those without having to understand the internal storage.

**Comment by dorranh** [comment]
@utdemir, I totally agree - it was a bit unfortunate having to write a traversal to be able to touch all of the slices. I think that @gkaracha created an issue related to this (https://github.com/tezos-checker/checker/issues/150). Perhaps we can centralize ideas for additional views there?

**Comment by gkaracha** [comment]
Ah, yes, that's a very good point @utdemir. Ideally that'd be the case for all our scenaria in the e2e tests I guess.

However: If the operations are just lookups in our storage we could offer them as views, but if they are calculations that we otherwise do not use elsewhere within checker it might not be a good idea to make views for those. They could get out of sync (and I expect advanced users will anyway look into the storage).

Either way I agree that #150 is a good place to keep all ideas for views, we should probably add the two you recommended to the list :+1:

**Pull request merged**